lib/nghttp2_hd.c

```diff
@@ -594,8 +594,20 @@ static void hd_map_remove(nghttp2_hd_map *map, nghttp2_hd_entry *ent) {
 static int hd_ringbuf_init(nghttp2_hd_ringbuf *ringbuf, size_t bufsize,
                            nghttp2_mem *mem) {
   size_t size;
-  for (size = 1; size < bufsize; size <<= 1)
+  size_t max_size;
+
+  max_size = SIZE_MAX / sizeof(nghttp2_hd_entry *);
```

Comment:
```suggestion
  const size_t max_size = SIZE_MAX / sizeof(nghttp2_hd_entry *);

```
Add const because this is compile time constant.